src/modules/teachers/components/AssignmentForm.jsx

```diff
@@ -309,13 +313,28 @@ class AssignmentForm extends Component {
         <div className="form-group">
           {this.renderSelectedSubjects()}
         </div>
-        <div className="form-group">
-          <p>Use the map to choose a set of images for your students to identify and "Select for assignment". </p>
-          <button className="btn btn-default" disabled={this.editMode()} onClick={this.selectNewSubjects}>Select images</button>
-        </div>
-        <div className="form-group">
-          <button type="submit" disabled={this.state.loading} className="btn btn-primary pull-right">Submit</button>
-        </div>
+        {(savedSubjectsIDs.length === 0 && !this.state.loading && !(this.props.fields && this.props.fields.subjects))
+          ? <div className="form-group">
+              <p>You need to select images for this assignment. Click "Select images" below, then use the map to choose a set of images for your students to identify and click "Select for assignment". </p>
+              <button className="btn btn-primary" disabled={this.editMode()} onClick={this.selectNewSubjects}>Select images</button>
+            </div>
+          : null
+        }
+        {(savedSubjectsIDs.length > 0 && !this.state.loading && !(this.props.fields && this.props.fields.subjects))
+          ? <div className="form-group">
+              <p>You've already selected images for this assignment. If you wish to replace them with newer images, click the "Select new images". Otherwise, click "Submit" to create the assignment.</p>
```

Comment:
I don't think we really need the first sentence "You've already selected images for this assignment." Or perhaps, just get rid of "already".